### ISSUE TITLE
[proposal] servicegraph support virtual/peer node 

### ISSUE THREAD

**Issue by JaredTan95** [open-telemetry/opentelemetry-collector-contrib#17196]
### Component(s)

processor/servicegraph

### Is your feature request related to a problem? Please describe.

as https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9232#issue-1202113403 designed, the current processor discards the call relationship when processing the following scenarios:

- It is requested by the browser to the backend, but the browser side does not integrate OTEL/sdk, and the backend integrates otel. So, the processor only received backend `server` span and will drop it from the cache when expired.

- The backend requests a service for an external address(like cloud service), and the backend integrates OTEL/sdk, but the external address does not integrate OTEL.

### Describe the solution you'd like

so, I propose to patch edge before the processor discards these spans, depending on the kind of span in the cache:

- extract remote peer attributes in `client` and `consumer` span in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0f57f296f4482ed0a8b01640053b231e63c5bda6/processor/servicegraphprocessor/processor.go#L201, and then store above attributes in `edge`.
- patch those edge before expired: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0f57f296f4482ed0a8b01640053b231e63c5bda6/processor/servicegraphprocessor/internal/store/store.go#L117
- if expired span's kind is `server`, we can directly patch `client=user`, `user` represents all of requert source, maybe browser or curl.etc

finally , the metrics geanerated would be like:

```bash
traces_service_graph_request_total{client="user",server="B"} 10

traces_service_graph_request_total{client="a",server="http://example.com/hello-world"} 1

traces_service_graph_request_total{client="a",server="grpc.hello_world"} 1

traces_service_graph_request_total{client="a",server="10.10.10.1:8080"} 1
```

**Comment by JaredTan95** [comment]
I already finished it and tested in my fork repo. if the proposal makes sense, I will contribute it upstream. :)

**Comment by mapno** [comment]
Hi! Thanks for the proposal, I like the two use-cases described here. They can be a nice addition to the servicegraph processor.

We can make the actual implementation a bit more generic, so it doesn't spill to the `store` pkg. We also need to make sure that the use-cases are included in the semantic conventions-ie. `net.sock.peer.addr` is meant to be used for those types of connections.

**Comment by JaredTan95** [comment]
> We also need to make sure that the use-cases are included in the semantic conventions-ie. `net.sock.peer.addr` is meant to be used for those types of connections.

After careful consideration, I found that not every span attribute has a generic tag like `net.sock.peer.addr'

**Comment by mapno** [comment]
> After careful consideration, I found that not every span attribute has a generic tag like net.sock.peer.addr.

Are there any attributes in the semantic conventions that we can reliably use to extract the data in the use-cases you described?

**Comment by JaredTan95** [comment]
> Are there any attributes in the semantic conventions that we can reliably use to extract the data in the use-cases you described?

Unfortunately, it seems that this does not exist. :-(

**Comment by djaglowski** [comment]
Closed by #17350
